village/src/App.js

```diff
@@ -1,25 +1,79 @@
 import React, { Component } from 'react';
+import axios from 'axios';
+import {Route, NavLink} from 'react-router-dom';
 
-import './App.css';
 import SmurfForm from './components/SmurfForm';
 import Smurfs from './components/Smurfs';
 
+import styled from 'styled-components';
+
+// Styled Components
+
+const StyledApp = styled.div`
+  text-align: center;
+`
+
+const NavBar = styled.nav`
+  display: flex;
+  justify-content: space-evenly;
+  background: #5D9530;
+  height: 50px;
+  align-items: center;
+  font-size: 1.2rem;
+  font-weight: bold;
+  
+  a {
+    text-decoration: none;
+    color: white;
+
+    &:hover {
+      transform: scale(1.1);
+      transition: all 200ms ease;
+    }
+  }
+`
+
+// App Component
+
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
       smurfs: [],
     };
   }
-  // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
-  // Notice what your map function is looping over and returning inside of Smurfs.
-  // You'll need to make sure you have the right properties on state and pass them down to props.
+
+  componentDidMount() {
+    axios.get('http://localhost:3333/smurfs')
+      .then(res => {
+        this.setState({smurfs: res.data})
+      })
+      .catch(err => {
+        console.log(err)
+      })
+  }
+
+  addSmurf = smurf => {
+    axios.post('http://localhost:3333/smurfs', smurf)
+      .then(res => {
+        console.log(res)
+        this.setState({smurfs: res.data})
+      })
+      .catch(err => {
+        console.log(err)
+      })
+  }
+
   render() {
     return (
-      <div className="App">
-        <SmurfForm />
-        <Smurfs smurfs={this.state.smurfs} />
-      </div>
+      <StyledApp>
+        <NavBar>
+          <NavLink to="/">Smurf Village</NavLink>
```

Comment:
To style an active NavLink, add a css styling to a `.active` class inside your NavBar styled component:
```
.active {
  color: blue;
}
```

and add `exact` to the Navlink path:
```
  <NavLink exact to="/">Smurf Village</NavLink>
```

This way, when you click Smurf Village on the Navbar, the text color will stay blue to indicate that you're currently on the Smurf Village route. If you click the Add New Smurf, then text color will toggle automatically.